context.go

```diff
@@ -1056,7 +1056,7 @@ func (c *Context) FileFromFS(filepath string, fs http.FileSystem) {
 // On the client side, the file will typically be downloaded with the given filename
 func (c *Context) FileAttachment(filepath, filename string) {
 	if isASCII(filename) {
-		c.Writer.Header().Set("Content-Disposition", `attachment; filename="`+filename+`"`)
+		c.Writer.Header().Set("Content-Disposition", `attachment; filename="`+strings.Replace(filename, "\"", "\\\"", -1)+`"`)
```

Comment:
This can fix this specific situation (that enables RFD attack) but what about using [filePath#Clean](https://pkg.go.dev/path/filepath#Clean)?

Reply:
I tried using that method, 
but the `"` was not escaped (or URL Encoded), so The vulnerability was not fixed.

Reply:
But since it's for file reading those quotes will be needed? File params would be readable with or without it, isn't?

Reply:
@fabricioereche 

Perhaps you are confusing the file name with the file path.
There may be some misunderstandings, so I will add to this.

--------------

The first thing to escape is the `filename`. Not `filepath`.

The second thing, The Filename escape requirements are then as follows
 * `"` --> `%2F`, `\"`
 * `\r` --> `%0D`, `\\r`
* `\n` --> `%0A`, `\\n`

The third thing, 
This line is the process of "escaping the 'file name at download time' on the HTTP Response Header.
In other words, filepath is irrelevant.

(The part that actually reads the file is the following line (`filepath`) )
https://github.com/gin-gonic/gin/pull/3556/files#diff-552f47512a00afe5fc6850cc9ddc830a6daeca162750e50aab4ed549685e0253R1063

---------------------

> But since it's for file reading those quotes will be needed? File params would be readable with or without it, isn't?

This is answered by the third matter mentioned above.
This tells the Request side to "download the file with this kind of file name".
It is not the file path to be read by the Server side.

Does this answer your question?